src/fides/api/ops/util/oauth_util.py

```diff
@@ -115,6 +132,19 @@ def verify_callback_oauth(
     return webhook
 
 
+async def get_root_client(
+    db: Session = Depends(get_db), client_id: str = CONFIG.security.oauth_root_client_id
```

Comment:
I see 🤔 

It doesn't feel good to me that we would be shipping something that we know is broken in a specific state, but here would be my justification:
* We have a ticket open for it, this is a known bug across the application and isn't specific to this use-case
* `Demo` mode should primarily be used by us to demo, and in that case we could take the extra step of adding a client (but then at that point why bother having a `demo` env?)
* Figuring out a solution that instead entirely removes auth as opposed to overwriting it would be required

I think we have two potential solutions here:
1. Remove the `demo` mode altogether, which also removes the bug. This is a new feature so we wouldn't be "losing" functionality
2. Ship as-is and document the caveat.

I'm leaning towards #1, but could understand #2 as well given that we are aware of this bug and are working to quash it.

Thoughts @pattisdr @seanpreston ?

Reply:
I'd lean towards `#1` too personally and we prioritize getting https://github.com/ethyca/fides/issues/1852 fixed first.  It would be worth revisiting if we needed client id persisted on these resources in the first place with the direction the product is going.